tlslite/tlsrecordlayer.py

```diff
@@ -912,3 +928,45 @@ def _changeWriteState(self):
 
     def _changeReadState(self):
         self._recordLayer.changeReadState()
+
+
+    def writeHeartbeat(self, type, payload, padding_length):
+        """Start a write operation of heartbeat_request.
+
+        @type payload_length: int
+        @param payload_length: Length of payload.
+
+        @type payload: bytes
+        @param payload: Payload, that we want send in request and
+                        get at response.
+
+        @raise socket.error: If a socket error occurs.
+        """
+        try:
+            if self.closed:
+                raise TLSClosedConnectionError(
+                    "attempt to write to closed connection")
+            heartbeat_request = Heartbeat().create(type, payload,
+                                                   padding_length)
+
+            for result in self._sendMsg(heartbeat_request, \
+                                        randomizeFirstBlock=True):
```

Comment:
`randomizeFirstBlock` will fragment the message, Heartbeat messages MUST NOT be fragmented

in general, since the fragmentation happens inside `_sendMsg`, I'm not convinced that using this method over `_sendMsgThroughSocket` is correct